### PR TITLE
Align item endpoint with items endpoint

### PR DESCRIPTION
**Pull request by spacemanspiff2007**
Add option to skip members, e.g. if the goal of the request to just get the metadata of a group item

### CODE REVIEW

**Comment by spacemanspiff2007** [comment]
I'm not sure why the build timed out when it was working fine before.
Could any one of the maintainers restart the CI Build please?

**Comment by cweitkamp** [comment]
Jenkins succeed, but the new build environment failed because of timezone issue:

```
Error:  Tests run: 19, Failures: 6, Errors: 0, Skipped: 0, Time elapsed: 0.432 s <<< FAILURE! - in org.openhab.core.thing.internal.profiles.TimestampOffsetProfileTest
Error:  org.openhab.core.thing.internal.profiles.TimestampOffsetProfileTest.testOnStateUpdateFromHandler(ParameterSet)[4]  Time elapsed: 0.011 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <+01:00> but was: <+02:00>
	at org.openhab.core.thing.internal.profiles.TimestampOffsetProfileTest.testOnStateUpdateFromHandler(TimestampOffsetProfileTest.java:149)
```

@wborn Any ideas?

**Review by cweitkamp — APPROVED** [review]
LGTM. Thanks.

**Comment by wborn** [comment]
> @wborn Any ideas?

It also fails on a local build for me now.
Probably the test needs to be fixed so it works regardless of any DST changes. :wink:

**Comment by cweitkamp** [comment]
Hm, okay. Then it was my fault as I added them back in the days.

File: bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/item/ItemResource.java [hunk 1]
```diff
@@ -229,6 +229,7 @@ public Response getItems(final @Context UriInfo uriInfo, final @Context HttpHead
     public Response getItemData(final @Context UriInfo uriInfo, final @Context HttpHeaders httpHeaders,
             @HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @Parameter(description = "language") @Nullable String language,
             @QueryParam("metadata") @Parameter(description = "metadata selector") @Nullable String namespaceSelector,
+            @DefaultValue("true") @QueryParam("recursive") @Parameter(description = "get member items if the item is a group item") boolean recursive,
```
[the remark this answers is not in the document]
Comment (spacemanspiff2007) [reply bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/item/ItemResource.java:232]: @cweitkamp What do you think? ``recursive`` or ``members``?
Imho ``members`` is more correct but on the items endpoint it's called ``recursive``,

[the remark this answers is not in the document]
Comment (cweitkamp) [reply bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/item/ItemResource.java:232]: From my pov "recursive" is okay as we are using it in the other method too. Keep in mind that it resolves nestled groups too.

**Comment by cweitkamp** [comment]
I will try to resolve the failing tests later.

**Review by cweitkamp — APPROVED** [review]

**Pull request merged**